src/credentials/FabricTable.cpp

```diff
@@ -751,6 +758,20 @@ CHIP_ERROR FabricTable::Delete(FabricIndex index)
     ReturnErrorOnFailure(err);
 
     ReleaseFabricIndex(index);
+
+    // If we ever start moving the FabricInfo entries around in the array on
+    // delete, we should update DeleteAllFabrics to handle that.
+
+    if (!mNextAvailableFabricIndex.HasValue())
+    {
+        // We have one available now.
+        mNextAvailableFabricIndex.SetValue(index);
```

Comment:
This will always make us use the last released fabric index. The spec calls for it to be monotonically increasing until you reach a free one, from the last allocated

Reply:
No, this will make us use the last released fabric index if we were in a state where:

1. CHIP_CONFIG_MAX_FABRICS is set to 254.
2. The table was full.

So that `mNextAvailableFabricIndex` could not have a meaningful value because there were no indices available at all.

In all other cases, removing a fabric does not affect `mNextAvailableFabricIndex`.